modules/internal/ChapelArray.chpl

```diff
@@ -2356,31 +2347,30 @@ module ChapelArray {
       compilerError("Cannot add indices to this domain type");
   }
 
-  proc +(i, d: domain) where isSubtype(i.type, index(d)) {
+  proc +(i, d: domain) where isSubtype(i.type, index(d)) && !isIrregularDom(d) {
```

Comment:
> I am unexcited about adding `&& !isIrregularDom(d)` because it is unnecessary.
> If you feel strongly about it, OK to leave it as-is. Otherwise please revert.

It was necessary to resolve an ambiguity caused by another change. I had to exclude irregular domains here, or else this overload would compete with another overload of equal specificity and it would be ambiguous.

Reply:
```chpl
var x = {1,2};
var y = 3 + x;
writeln(y.sorted());
```
Without this change:
```
$ ./bin/darwin-x86_64/chpl foo.chpl
foo.chpl:1: In module 'foo':
foo.chpl:2: error: ambiguous call '+(3, DefaultAssociativeDom(int(64),true))'
$CHPL_HOME/modules/internal/ChapelBase.chpl:1938: note: candidates are: +(a: uint(64), b: int(64)) [16674]
$CHPL_HOME/modules/internal/ChapelBase.chpl:1959: note:                 +(param a: int(64), b: int(64)) [1151577]
$CHPL_HOME/modules/internal/ChapelBase.chpl:266: note:                 +(a: int(64), b: int(64)) [458737]
$CHPL_HOME/modules/internal/ChapelBase.chpl:268: note:                 +(a: real(64), b: real(64)) [458869]
$CHPL_HOME/modules/internal/ChapelBase.chpl:270: note:                 +(a: complex(128), b: complex(128)) [458957]
$CHPL_HOME/modules/internal/ChapelBase.chpl:274: note:                 +(a: real(64), b: complex(128)) [459154]
$CHPL_HOME/modules/internal/ChapelBase.chpl:275: note:                 +(a: complex(128), b: real(64)) [459237]
$CHPL_HOME/modules/internal/ChapelArray.chpl:2350: note:                 +(i: int(64), d: DefaultAssociativeDom(int(64),true)) [1406672]
$CHPL_HOME/modules/internal/ChapelArray.chpl:2362: note:                 +(i: int(64), d: DefaultAssociativeDom(int(64),true)) [1406760]
foo.chpl:2: note: unresolved call had id 207098
```
With this change to avoid ambiguity:
```
$ ./bin/darwin-x86_64/chpl foo.chpl
/Users/leekillough/chapel $ ./foo
1 2 3
```
Notice that the condition is complementary to another one already in master:

```chpl
 proc +(i, d: domain) where isSubtype(i.type, index(d)) && !isIrregularDom(d) {
    if isRectangularDom(d) then
      compilerError("Cannot add indices to a rectangular domain");
    else
      compilerError("Cannot add indices to this domain type");
  }

// ...

  proc +(i, in d: domain) where isSubtype(i.type,index(d)) && isIrregularDom(d) {
    d.add(i);
    return d;
  }
```

Reply:
Oh I see it now. It is not complementary to the one in master, however you are right, it is essential.
Please disregard my request.

Reply:
I think the change of the second overload to use `in` intent for the `d` formal, caused the ambiguity. Before changing it to use `in`, the two functions had the same exact formal signature, and one was more "specific" in its constraints than the other, so there was no ambiguity. Now that the two signatures differ slightly, with one using default `const ref` intent and the other using `in` intent, they don't have the same signature and can't be disambiguated solely based on the specificity of their constraints. We have to change the first overload to exclude the `isIrregularDomain()` constraint of the second overload, so that their constraints never coincide.